recipes/ncurses/all/conanfile.py

```diff
@@ -0,0 +1,203 @@
+from conans import AutoToolsBuildEnvironment, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+from contextlib import contextmanager
+import os
+
+
+class NCursesConan(ConanFile):
+    name = "ncurses"
+    description = "The ncurses (new curses) library is a free software emulation of curses in System V Release 4.0 (SVr4), and more"
+    topics = ("conan", "ncurses", "terminal", "screen", "tui")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.gnu.org/software/ncurses"
+    license = "X11"
+    exports_sources = "patches/**"
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "pkg_config"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_cxx": [True, False],
+        "with_pcre2": [True, False],
+        "with_reentrant": [True, False],
+        "with_widec": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_cxx": True,
+        "with_pcre2": False,
+        "with_reentrant": False,
+        "with_widec": False,
+    }
+
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.compiler == "Visual Studio":
+            if self.options.with_widec:
+                raise ConanInvalidConfiguration("with_widec is unsupported for Visual Studio")
+            raise ConanInvalidConfiguration("Unsupported on Visual Studio")
```

Comment:
I think you can just leave this one, and remove two lines above, as soon as Visual Studio isn't supported

Reply:
Yes I can, but would prefer to keep it as the `with_widec` fails to build because of missing symbols. So when somebody who want to enable MSVC sees that `with_widec` is a special case.
I should add this add a comment.